public/expressions/overlay_anomalies.ts

```diff
@@ -103,6 +103,56 @@ const convertAnomaliesToPointInTimeEventsVisLayer = (
   } as PointInTimeEventsVisLayer;
 };
 
+const checkIfPermissionErrors = (error): boolean => {
+  return typeof error === 'string'
+    ? error.includes(NO_PERMISSIONS_KEY_WORD) ||
+        error.includes(DOES_NOT_HAVE_PERMISSIONS_KEY_WORD)
+    : get(error, 'message', '').includes(NO_PERMISSIONS_KEY_WORD) ||
+        get(error, 'message', '').includes(DOES_NOT_HAVE_PERMISSIONS_KEY_WORD);
+};
+
+//Helps convert any possible errors into either permission, deletion or fetch related failures
+export const getVisLayerError = (error): VisLayerError => {
+  let visLayerError: VisLayerError = {} as VisLayerError;
+  if (checkIfPermissionErrors(error)) {
+    visLayerError = {
+      type: VisLayerErrorTypes.PERMISSIONS_FAILURE,
+      message:
+        error === 'string'
+          ? error
+          : error instanceof Error
+          ? get(error, 'message', '')
+          : '',
+    };
+  } else if (
+    typeof error === 'string'
+      ? error.includes(DETECTOR_HAS_BEEN_DELETED)
+      : get(error, 'message', '').includes(DETECTOR_HAS_BEEN_DELETED)
```

Comment:
nit: you could pull this out into a helper fn like you do for `checkIfPermissionErrors()` - something like `checkIfDeletionErrors()`